### PR TITLE
Add option to process all detected faces within a frame

### PR DESCRIPTION
**Pull request by CeeBeeEh**
I've added the ability to swap all of the faces in a frame. This is useful of you want to face swap a video with multiple people within it. It adds the ability to edit the resulting video in a NLE without having to preprocess the video to isolate each face.

### CODE REVIEW

**Comment by henryruhs** [comment]
A mind blowing addition for VR videos! We have to check your branch before merging it.

**Comment by CeeBeeEh** [comment]
I guess the use case works for VR. My intention was to faceswap a longer video (like a movie) and then isolate the swapped face in an editor. Instead of isolating first.

My reasoning is to do multiple faces and then editing together at once.

**Comment by henryruhs** [comment]
Please resolve the conflicts... can you provide an example file?

**Comment by henryruhs** [comment]
I don't want to be too picky but please align the checkboxes... they look wild :D

**Comment by CeeBeeEh** [comment]
@henryruhs It looks good on my end. I'm guessing you're using Windows? Python on Windows has a lot of inconsistencies when compared to all other platforms. I can try to make it look consistent on Windows, but I don't have a Windows system to test with.

roop_alignment

**Comment by CeeBeeEh** [comment]
@henryruhs merge the fix that "should" fix it, if not then I don't have a way to test the appearance and can't fix the UI elements on my end

**Comment by Mayorc1978** [comment]
SimSwap provides a method to identify different faces (creating an identity vector for each face) and use a method to match a face to another face within an identity threshold.
@CeeBeeEh Could you please explain what do you mean by VR Mode and NLE references in your post?

**Comment by CeeBeeEh** [comment]
@Mayorc1978 I didn't set the title to VR. I believe it was done to match up with this enhancement request #178 

As for the NLE comment, I realize after the fact that my comment is confusing. I simply meant it would make putting together a video with multiple swaps simpler by editing together the resulting videos in something like Kdenlive or After Effects.

**Comment by henryruhs** [comment]
Okay, so there was a misunderstanding? Can you please provide an example before and after video?

Merge conflicts appear again... sorry

**Comment by CeeBeeEh** [comment]
@henryruhs I resolved the conflicts and I think I found a solution for the checkbox alignment issue that should work across all platforms.

As for a before/after video, it would just be all the faces in a video being swapped.

The idea is this:
Souce: video with multiple people, I want to swap each one to a different person (let's say three individuals)

I have two options:
Edit the source video three separate times to mask out all faces except for one person. Run Roop for each video/face. Edit the results back into a single video for desired result.

Better (IMO) option:
Swap all the faces in source video. So run the video for each target face. Pull all the results into editor and mask out each face I want in the final video.

Basically this merge gives two options: swap all faces just for a fun video, or simpler workflow to do multi-person face swaps.

File: core/swapper.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import cv2
 import insightface
 import core.globals
-from core.analyser import get_face
+from core.analyser import get_face, get_all_faces
```
Comment (henryruhs) [inline core/swapper.py:6]: can we rename it to get_one_face() and get_many_faces()

File: core/swapper.py [hunk 2]
```diff
@@ -21,14 +21,29 @@ def process_video(source_img, frame_paths):
...
             try:
-                face = get_face(frame)
-                if face:
-                    result = get_face_swapper().get(frame, face, source_face, paste_back=True)
+                if core.globals.all_faces:
+                    all_faces = get_all_faces(frame)
+                    result = frame
+                    if len(all_faces) > 0:
+                        for singleFace in all_faces:
+                            if singleFace:
```
Comment (henryruhs) [inline core/swapper.py:31]: I think if you rewrite the code a bit you can extract a method here that contains the P and S conditions

can be reused in the one face block aswell

File: core/globals.py [hunk 3]
```diff
@@ -1,7 +1,8 @@
...
-use_gpu = False
-providers = onnxruntime.get_available_providers()
-
-if 'TensorrtExecutionProvider' in providers:
-    providers.remove('TensorrtExecutionProvider')
+import onnxruntime
+
+use_gpu = False
+providers = onnxruntime.get_available_providers()
+all_faces = False
```
Comment (henryruhs) [inline core/globals.py:5]: all boolean to the top

rename to many_faces

File: core/swapper.py [hunk 4]
```diff
@@ -21,14 +21,29 @@ def process_video(source_img, frame_paths):
     with tqdm(total=len(frame_paths), desc="Processing", unit="frame", dynamic_ncols=True, bar_format='{l_bar}{bar}| {n_fmt}/{total_fmt} [{elapsed}<{remaining}, {rate_fmt}{postfix}]') as progress:
         for frame_path in frame_paths:
             frame = cv2.imread(frame_path)
+            swapper = get_face_swapper()
             try:
-                face = get_face(frame)
-                if face:
-                    result = get_face_swapper().get(frame, face, source_face, paste_back=True)
+                if core.globals.all_faces:
+                    all_faces = get_all_faces(frame)
```
Comment (henryruhs) [inline core/swapper.py:27]: use wording many_faces here

**Comment by henryruhs** [comment]
Thanks, I made a proper code review... hopefully not too strict for a POC but we like to move forward to clean code

**Comment by CeeBeeEh** [comment]
"set target branch to next"

Done

**Comment by CeeBeeEh** [comment]
Made the requested changes. I went with "get_face_single" and "get_face_many" to make it consistent and to read a bit better. I also refactored the swapper.py file to split the code into smaller functions and to create fewer copies of the frame.

**Comment by Gitterman69** [comment]
is this working now? multi face support is enabled?

**Comment by suphamster** [comment]
If video contains for example man and woman faces and I wanna to replace woman's only is it possible for now?
